scripts/azureml-assets/azureml/assets/release/model_publish_utils/model_publish_utils.py

```diff
@@ -0,0 +1,116 @@
+import os
+import shutil
+import yaml
+import logging
+from typing import List
+from azureml.assets.util import logger
+import azureml.evaluate.mlflow as mlflow
+from mlflow.models import ModelSignature
+from sqlalchemy import true
+from transformers import  AutoTokenizer, AutoConfig, pipeline
+from transformers import (
+    AutoModelForSequenceClassification,
+    AutoModelForMaskedLM,
+    AutoModelForTokenClassification,
+    AutoModelForQuestionAnswering,
+    AutoModelWithLMHead
+)
+
+class ModelPublishUtils:
+
+    @classmethod
+    def __init__(self, spec_path: str) -> None:
+        
+        self.model_yaml_path = spec_path + "/model.yml"
+        self.asset_yaml_path = spec_path + "/asset.yml"
+        self.model_name = self._get_model_name()
+        self.model_uri = self._get_model_uri()
+        self.model_commit = self._get_model_commit()
+        self.model_task_name = self._get_model_task_name()
+        self.model_dir_name = spec_path + self.model_name
+
+    @classmethod
+    def _get_model_name(self) -> str:
+        with open(self.model_yaml_path) as f:
+            model_file = yaml.safe_load(f)
+        return model_file["name"]
+
+    @classmethod
+    def _get_model_uri(self) -> str:
+        with open(self.asset_yaml_path) as f:
+            asset_file = yaml.safe_load(f)
+        return asset_file["package"]["path"]
+
+    @classmethod
+    def _get_model_commit(self) -> str/None:
+        with open(self.asset_yaml_path) as f:
+            asset_file = yaml.safe_load(f)
+        return asset_file["package"]["sha"] if asset_file["package"]["sha"] else None
+    
+    @classmethod
+    def _get_model_task_name(self) -> str:
+        with open(self.asset_yaml_path) as f:
+            asset_file = yaml.safe_load(f)
+        return asset_file["task"]
+
+    @classmethod
+    def _git_clone_model(self) -> None:
+        if self.validate_model_and_task():
+            cmd = f'git clone {self.model_uri} {self.model_dir_name}'
+            os.system(cmd)
+            if self.model_commit:
+                os.chdir(f'{self.model_dir_name}')
+                os.system(f'git reset --hard {self.model_commit}')
+
+    # Convert Model to ISV FLavoured MLFlow model and save it.
+    @classmethod
+    def _save_mlflow_model(self):
+        config = AutoConfig.from_pretrained(self.model_name)
+        misc_conf = {"task_type": self.model_task_name}
+        task_model_mapping = {
+            "multiclass": AutoModelForSequenceClassification,
```

Comment:
these are too specific to your scenario, can you make it generic? This toolset directory will be used by lots of folks (both internal and external) to reuse and see how to publish assets, we don't want **any** asset specific content here.

Reply:
for example, in code here, you should handle conversion in a generic way, the yaml config could specify the transformation needed

Reply:
Extracted it to a new class called model_publsih_utils